modules/client/src/internal/client/methods.ts

```diff
@@ -281,12 +285,16 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     params: DepositParams,
   ): AsyncGenerator<DaoDepositStepValue> {
     switch (params.type) {
-      case TokenType.ERC20 || TokenType.NATIVE:
+      case TokenType.NATIVE:
+      case TokenType.ERC20:
```

Comment:
Different functions should exist here. The function `depositErc20(params)` should not handle deposits of native tokens.